.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,41 @@
+name: clang-format
+
+on:
+  # Trigger the workflow in the following cases:
+  #  -- After a pull request to the 'development' branch; these changes will
+  #     later be integrated into 'main' when it is merged in.
+  #  -- After a push to any branch except 'main' and 'development', as these
+  #     branches should never be updated directly by a user.
+  pull_request:
+    branches:
+      - development
+  push:
+    branches-ignore:
+      - main
+      - development
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    # Need to checkout the repository and specify the ref in the case of non-
+    # push events for the git-auto-commit-action (at the end) to work correctly.
+    # See: https://github.com/stefanzweifel/git-auto-commit-action
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+
+    # Format all files in the src/ directory except cfortran.h, lapack_qz.h,
+    # superlu.c, superlu_dist.c, which sit in the src/generic/ subdirectory.
+    - uses: PuneetMatharu/clang-format-lint-action@v0.13
+      with:
+        files: ^(src)\/(?!.*(cfortran|lapack_qz))(.+)\.(h|cc)$
+        clangFormatVersion: 12
+        style: file
+        inplace: True
+
+    - uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
Only glanced through this. This probably means that you're using an "action" defined by somebody else. What if this somebody else breaks/removes that action? Would it be safer to copy it across?

Reply:
Yes, it does. It's a well established repository so I would be very surprised if it breaks (although it is not impossible). The authors also use a semantic versioning system, so it should not break anything for us as long as we stick to a specific major version (which is why is what the `@4` is used to specify). _If_ it ever breaks, we can just add a hotfix to update the workflow to use the most recent non-breaking minor version. 

If you definitely want us to have our own forked copy, it's worth bearing in mind that our other workflow scripts also use actions that other people have published, so if we're going to copy things over, we might as well copy all of them over. It's worth running this by @jondea too if so.